### ISSUE TITLE
get_module_attribute() returns '' if module is not installed

### ISSUE THREAD

**Issue by htgoebel** [pyinstaller/pyinstaller#3483]
Follow up to https://github.com/pyinstaller/pyinstaller/issues/3428#issuecomment-384111742

[get_module_attribute](https://github.com/pyinstaller/pyinstaller/blob/v3.3.1/PyInstaller/utils/hooks/__init__.py#L261]) basically does this:
- return an empty string if the module was not imported
- return the attribute value if the attribute was defined
- The intended behavior if import fails is not documented (see doc-string)

```
>>> from PyInstaller.utils.hooks import get_module_attribute
>>> a = get_module_attribute('sdahjkahdfjk', '__version__')
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ImportError: No module named sdahjkahdfjk
>>> a
''
>>> a = get_module_attribute('string', 'non_existing_attribute')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "PyInstaller/utils/hooks/__init__.py", line 298, in get_module_attribute
    'Module %r has no attribute %r' % (module_name, attr_name))
AttributeError: Module 'string' has no attribute 'digits____'
```

Todo:
- [ ] decide on intended behavior of `get_module_attribute` if module can't be imported. I suggest making this raise an ImportError.
  A quick `git grep` showed that this will break [some test](https://github.com/pyinstaller/pyinstaller/blob/v3.3.1/tests/functional/test_libraries.py#L280), which should be easy to work around. Beside this test `get_module_attribute` is only used in hooks, thus this should be save. `get_module_file_attribute` is already [expected to raise an ImportError](https://github.com/pyinstaller/pyinstaller/blob/v3.3.1/tests/unit/test_hookutils.py#L256) if the module is missing.
  For making this decision, one could first implement and look what the outcome is :-)
- [ ] Implement test-case (in tests/unit/test_hookutils.py)
- [ ] Implement (print some `attr_value_if_not_imported` value) and update functions doc-string
- [ ] Check if anything else breaks.

**Comment by MagnumOpus21** [comment]
Ok @htgoebel , here's the problem. Python3.6 throw ModuleNotFoundError while, any version less than that will throw an ImportError. So, here's what I am planning to do. I will check for any error that is thrown when the import is attempted. I'll catch it check if it is any one of the two. If it is, I'll raise an Import error. If not, if we can import the package, but cannot find any attribute, the previous behavior will remain as it is. Seems fair enough?

**Comment by htgoebel** [comment]
`ModuleNotFoundError` is a subclass of `ImportError`, see https://docs.python.org/3/library/exceptions.html#Exception

**Comment by bwoodsend** [comment]
Fixed by 169c582a2e567dff32e293013bdfdedee3fa7dc2.
